plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -0,0 +1,305 @@
+  ## Limit OpenWeatherMap query interval. See calls per minute info at: https://openweathermap.org/price
+  interval = "10m"
+`
+
+func (n *OpenWeatherMap) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *OpenWeatherMap) Description() string {
+	return "Read current weather and forecasts data from openweathermap.org"
+}
+
+func (n *OpenWeatherMap) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+	var strs []string
+
+	base, err := url.Parse(n.BaseUrl)
+	if err != nil {
+		return err
+	}
+
+	// Create an HTTP client that is re-used for each
+	// collection interval
+
+	if n.client == nil {
+		client, err := n.createHttpClient()
+		if err != nil {
+			return err
+		}
+		n.client = client
+	}
+	units := n.Units
+	if units == "" {
+		units = defaultUnits
+	}
+	for _, fetch := range n.Fetch {
+		if fetch == "forecast" {
+			var u *url.URL
+			var addr *url.URL
+
+			for _, city := range n.CityId {
+				u, err = url.Parse(fmt.Sprintf("/data/2.5/forecast?id=%s&APPID=%s&units=%s", city, n.AppId, units))
+				if err != nil {
+					acc.AddError(fmt.Errorf("Unable to parse address '%s': %s", u, err))
+					continue
+				}
+				addr = base.ResolveReference(u)
+				wg.Add(1)
+				go func(addr *url.URL) {
+					defer wg.Done()
+					acc.AddError(n.gatherUrl(addr, acc, true))
+				}(addr)
+			}
+		} else if fetch == "weather" {
+			j := 0
+			for j < len(n.CityId) {
+				var u *url.URL
+				var addr *url.URL
+				strs = make([]string, 0)
+				for i := 0; j < len(n.CityId) && i < owmRequestSeveralCityId; i++ {
+					strs = append(strs, n.CityId[j])
+					j++
+				}
+				cities := strings.Join(strs, ",")
+
+				u, err = url.Parse(fmt.Sprintf("/data/2.5/group?id=%s&APPID=%s&units=%s", cities, n.AppId, units))
+				if err != nil {
+					acc.AddError(fmt.Errorf("Unable to parse address '%s': %s", u, err))
+					continue
+				}
+
+				addr = base.ResolveReference(u)
+				wg.Add(1)
+				go func(addr *url.URL) {
+					defer wg.Done()
+					acc.AddError(n.gatherUrl(addr, acc, false))
+				}(addr)
+			}
+
+		}
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (n *OpenWeatherMap) createHttpClient() (*http.Client, error) {
+
+	if n.ResponseTimeout.Duration < time.Second {
+		n.ResponseTimeout.Duration = defaultResponseTimeout
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{},
+		Timeout:   n.ResponseTimeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (n *OpenWeatherMap) gatherUrl(addr *url.URL, acc telegraf.Accumulator, forecast bool) error {
+	resp, err := n.client.Get(addr.String())
+
+	if err != nil {
+		return fmt.Errorf("error making HTTP request to %s: %s", addr.String(), err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("%s returned HTTP status %s", addr.String(), resp.Status)
+	}
+	contentType := strings.Split(resp.Header.Get("Content-Type"), ";")[0]
+	switch contentType {
+	case "application/json":
+		err = gatherWeatherUrl(bufio.NewReader(resp.Body), forecast, acc)
+		return err
```

Comment:
nit: just `return gatherWeatherUrl(bufio.NewReader(resp.Body), forecast, acc)` instead of assigning it to `err` first.